src/openfermion/utils/_lcu_util.py

```diff
@@ -0,0 +1,181 @@
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+"""Utility methods for LCU circuits."""
+
+from __future__ import absolute_import
+from __future__ import division
+
+import math
+
+
+def _partial_sums(vals):
+    total = 0
+    for v in vals:
+        yield total
+        total += v
+    yield total
+
+
+def _differences(weights):
+    p = None
+    have_p = False
```

Comment:
I know what p and have_p are because I know how this is all supposed to work. But it might be good to have a little more description of what this function is doing and what p / have_p are so somebody looking at this code without our paper could have a chance at figuring out what is going on,.

Reply:
Done.